From 0094dea49d8bc5a60c044da1e01417c29397c7b0 Mon Sep 17 00:00:00 2001
From: Dylan Van Assche <me@dylanvanassche.be>
Date: Sat, 17 Apr 2021 15:50:07 +0200
Subject: [PATCH 3/8] at: fast/poweroff is only available in newer firmware
 versions

Do not retry the AT command of the fast/poweroff setting as it may not be supported by the firmware
---
 src/at.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/at.c b/src/at.c
index a34025f..4469d26 100644
--- a/src/at.c
+++ b/src/at.c
@@ -215,7 +215,11 @@ static gboolean modem_response(gint fd,
             suspend_inhibit(manager, TRUE, TRUE);
             manager->modem_state = EG25_STATE_STARTED;
         }
-        else if (strstr(response, "ERROR"))
+        /*
+         * QCFG="fast/poweroff" configuration is only available in
+         * newer firmware versions
+         */
+        else if (strstr(response, "ERROR") && !strstr(response, "fast/poweroff"))
             retry_at_command(manager);
         else if (strstr(response, "OK"))
             process_at_result(manager, response);
-- 
2.31.1

